src/5.0_to_5.1/migrateDashboard.ts

```diff
@@ -40,6 +40,7 @@ export const migrateDashboard: MigrateDashboardCallback<
     dashboardState,
     (widgetState) =>
       migrateWidget(widgetState, {
+        supportCalculatedMeasuresMigration: true,
```

Comment:
I would like to avoid this, and use whether `namesOfCalculatedMeasuresToMigrate` and `measureToCubeMapping` are defined in order to know whether calculated measures should be migrated:

```
type OptionsForCalculatedMeasuresRemoval = {
  namesOfCalculatedMeasuresToMigrate: string[];
  measureToCubeMapping: { [measureName: string]: string[] };
};

/**
  * If calculated measures don't need to be migrated,
  * then the caller does not need to pass options for calculated measures removal.
  */
type Options = OptionsForCalculatedMeasuresRemoval | {};

/**
  * Returns whether `options` allows to remove saved calculated measures from the migrated `widgetState`.
  * This is done when migrating a Content Server, but not a notebook.
  */
function shouldMigrateCalculatedMeasures(options:  Options): options is  OptionsToMigrateCalculatedMeasures {
  return "namesOfCalculatedMeasuresToMigrate" in options && "measureToCubeMapping" in options;
}

export const migrateWidget: MigrateWidgetCallback<
  AWidgetState50,
  AWidgetState51,
  {
    namesOfCalculatedMeasuresToMigrate: string[];
    measureToCubeMapping: { [measureName: string]: string[] };
  } | {}
> = (options) => {
  if (shouldMigrateCalculatedMeasures(options)) {
    migrateCM();
  }
}
```

I prefer doing it this way, because even though it makes the function internals more verbose, it allows a leaner syntax for its callers: they don't have to pass the additional `supportCalculatedMeasuresMigration` argument, and `migrateNotebook` does not have to pass an empty `namesOfCalculatedMeasuresToMigrate` and an empty `measureToCubeMapping` which could otherwise seem obscure to readers who are not yet familiar with `migrateContentServer`.